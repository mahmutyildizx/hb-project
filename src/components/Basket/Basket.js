import React from "react";
import styles from "./Basket.module.scss";

function Basket({ totalQuantity }) {
  return (
    <div className={styles.basketContainer}>
      <span className={styles.count}>{totalQuantity() || 0}</span>
      <p>Sepetim</p>
    </div>
  );
}

export default Basket;
