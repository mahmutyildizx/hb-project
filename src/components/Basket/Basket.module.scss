.basketContainer {
  width: 100%;
  min-width: 120px;
  padding: 14px 20px;
  background-color: #ffffff;
  color: #b0b0b0;
  border: 1px solid #b0b0b0;
  border-radius: 4px;
  font-size: 16px;
  font-weight: 500;
  line-height: 24px;
  position: relative;
  cursor: pointer;
  .count {
    position: absolute;
    z-index: 2;
    width: 20px;
    height: 20px;
    border-radius: 50%;
    padding: 10px;
    top: -10px;
    right: -10px;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: #ff6000;
    color: #ffffff;
    font-size: 12px;
  }
  p {
    margin: 0;
    padding: 0;
    text-align: center;
    user-select: none;
  }
}
