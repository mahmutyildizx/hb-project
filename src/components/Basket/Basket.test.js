import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import preloadAll from "jest-next-dynamic";
import Basket from "./Basket";

afterEach(cleanup);

beforeAll(async () => {
  await preloadAll();
});

test("should render", () => {
  const { container } = render(<Basket />);
  expect(container.firstChild).toHaveClass("basketContainer");
});

test("should match snapshot", () => {
  const { container } = render(<Basket />);
  expect(container).toMatchSnapshot();
});

test("Basket label rendered as text", () => {
  render(<Basket totalQuantity="5" />);
  expect(screen.getByText("5")).toBeInTheDocument();
});
