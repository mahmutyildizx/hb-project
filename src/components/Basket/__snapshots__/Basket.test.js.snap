// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="basketContainer"
  >
    <span
      class="count"
    >
      0
    </span>
    <p>
      Sepetim
    </p>
  </div>
</div>
`;
